tests/trainer/test_trainer.py

```diff
@@ -684,7 +684,6 @@ def predict_step(self, batch, *_):
 
     trainer_fn = getattr(trainer, fn)
     path_attr = f"{fn}{'d' if fn == 'validate' else 'ed'}_ckpt_path"
-    assert getattr(trainer, path_attr) is None
```

Comment:
yup. since I removed these lines of code:
```
validated_ckpt_path: Optional[str] = None
tested_ckpt_path: Optional[str] = None
predicted_ckpt_path: Optional[str] = None
```
we don't initialize these variables to None, rather they are just set to their values at the relevant points in the code

Reply:
@daniellepintz this is a breaking change and needs to be reverted

```python
trainer = Trainer()
print(trainer.validated_ckpt_path)
>>> AttributeError: 'Trainer' object has no attribute 'validated_ckpt_path'
```

It should be defined in the `__init__`